### DISCUSSION TITLE
Regarding network protocol

### DISCUSSION THREAD

**Discussion by bardale** [finos/TimeBase-CE#272] [Q&A] [answered]
Greetings @alex-karpovich 
Can we use any other protocol (maybe http) instead of dxtick to connect timebase server using java timebase client library?

Regards,
Deepak

**Comment by alex-karpovich** [comment, marked as answer]
TimeBase Server has http protocol, but it's used by C++ and Python API.
I would suggest to use TimeBase Web Gateway RestFull API via http(s)/websockets

https://github.com/epam/TimebaseWS

**Reply by bardale** [reply]
Thanks @alex-karpovich
And is there any limitation for master/slave schema with replication with respect to protocol?  Does it only support dxtick in replication script?

**Reply by alex-karpovich** [reply]
Build-in replication works only with dxtick protocol.

Please explain your goal - I will try to help

**Reply by bardale** [reply]
We are exploring Cluster as well Master Slave architecture.  As per TB suggestion Master Slave  schema with replication has better latency.  So we want to setup Master Slave schema with replication.
Also we need your suggestion on https://github.com/finos/TimeBase-CE/discussions/264 to proceed.
